### PR TITLE
Recent changes

### PR DESCRIPTION
**Pull request by DavidMcCann-NOC**
- X-band .mat file reader and colocation tool
- Translation of Karlus' incidence angle script (although it turns out we may have an outdated version now, so this may need updating in the future)
- Docstring standardisation and Sphinx setup files for automatic documentation. Now working but will need some HTML style formatting

### CODE REVIEW

File: seastar/oscar/level1.py [hunk 1]
```diff
@@ -224,10 +224,160 @@ def compute_incidence_angle(ds):
...
+            + (N - gridinfo[1]) * np.sin(gridinfo[8])
+    orb_z = ds.OrbHeightImage
+    xlen = len(x_axis)
+    ylen = len(y_axis)
+
+#speedup_factor_x = 50
+#speedup_factor_y = 50
+    speedup_factor_x = 48
+    speedup_factor_y = 40
+    orbx = xr.DataArray(data=orb_x,
```
Comment (ACHMartin) [inline seastar/oscar/level1.py:283]: just for your information, no need to change the code, you can certainly do coarsen/rolling mean directly in numpy without the need to call for xarray. It is not useful here and it makes it more difficult to read/understand why/how xarray is used. (my understanding is it is only used for coarsen it)

File: seastar/oscar/level1.py [hunk 2]
```diff
@@ -224,10 +224,160 @@ def compute_incidence_angle(ds):
     nadir (radians) for each pixel
 
     """
-    X, Y = np.meshgrid(ds.CrossRange, ds.GroundRange, indexing='ij')
-    ds['IncidenceAngleImage'] = np.degrees(np.arctan(np.sqrt(2) *
-                                                     Y / ds.OrbHeightImage))
```
Comment (ACHMartin) [inline seastar/oscar/level1.py:244]: perhaps the old function could be renamed:
compute_incidence_angle_from_simple_geometry or something more explicit than simple_geometry

the other one might be called:
compute_incidence_angle_from_GBPGridInfo

File: seastar/oscar/level1.py [hunk 3]
```diff
@@ -1,10 +1,7 @@
 # -*- coding: utf-8 -*-
 """
-Functions to calculate varirous L2 data products for the OSCAR instrument.
```
[the remark this answers is not in the document]
Comment (DavidMcCann-NOC) [reply seastar/oscar/level1.py:3]: I had to stare at that for an inordinately long time to notice!

File: seastar/oscar/level1.py [hunk 4]
```diff
@@ -327,8 +326,48 @@ def compute_radial_surface_velocity(ds):
 
     return ds
 
+def compute_radial_surface_current(level1, aux, gmf='mouche12'):
```
[the remark this answers is not in the document]
Comment (DavidMcCann-NOC) [reply seastar/oscar/level1.py:496]: I also need to learn what **kwargs really are and how to use them properly to be honest

File: seastar/oscar/level1.py [hunk 5]
```diff
@@ -224,10 +224,160 @@ def compute_incidence_angle(ds):
...
+        inc_angle = xr.DataArray(data=inc_angle,
+                                 coords=[xaxis, yaxis],
+                                 dims=('CrossRange', 'GroundRange')
+                                 )
+        look_angle = xr.DataArray(data=look_angle,
+                                  coords=[xaxis, yaxis],
+                                  dims=('CrossRange', 'GroundRange')
+                                  )
+
+    ds['IncidenceAngleImage'] = inc_angle.interp_like(ds.CrossRange)
```
[the remark this answers is not in the document]
Comment (DavidMcCann-NOC) [reply seastar/oscar/level1.py:397]: Not sure If what i did was right or not but it comes out as we would expect (minus some edge effects from the NaN inpainting). This should be adapted to scrub the mention of CrossRange though, this was just a quick fix to get it running

File: seastar/oscar/level1.py (hunk 1 above)
Comment (DavidMcCann-NOC) [reply seastar/oscar/level1.py:283]: Thanks - I agree, it does make it a little more unnecessarily complicated. I like the coarsen functionality though so in the future i'll get it working directly in numpy

Comment (ACHMartin) [reply seastar/oscar/level1.py:283]: if there is already an xarray dataset it is fine to work directly on it, just here I wonder why we had a dataset and not using it after

**Pull request merged**